src/movepick.cpp

```diff
@@ -223,6 +223,7 @@ Move MovePicker::next_move() {
     case QSEARCH_TT :
     case PROBCUT_TT :
         ++stage;
+        cur = moves + 1;
```

Comment:
@pb00068  Can you give a hint why this had to be added?

Reply:
this was necessary to have deterministic behavior for method MovePicker::mark_current_illegal,
otherwise mark_current_illegal might act on uninitialized cur pointer.
FYI: There's a PR by nonlinear underway that simplifies away  MovePicker::mark_current_illegal, so also this instruction will be obsolete.